signup/tool/src/java/org/sakaiproject/signup/tool/jsf/SignupUIBaseBean.java

```diff
@@ -738,11 +754,13 @@ public String getAllAttendeesEmailAddressesFormatted() {
 		Set<String> emails = new HashSet<String>();
 		
 		StringBuilder sb = new StringBuilder();
-		for (TimeslotWrapper tsWrapper : timeslotWrappers) {
-			for(AttendeeWrapper atWrapper : tsWrapper.getAttendeeWrappers()) {
-				String email = atWrapper.getAttendeeEmail();
-				if(StringUtils.isNotBlank(email)){
-					emails.add(email);
+		if (timeslotWrappers!=null){

```

Comment:
I'm guessing that this check is needed because otherwise we get a NPE here, but it seems a little brittle. In that it's easy for some other change in the signup tool to mean that the NPE happens at a different point in the tool and then you've got this non-working code in signup that people aren't sure what cases is there for.